content/ideas/quickcheck-state-machine.md

```diff
@@ -0,0 +1,44 @@
+---
+title: Property-based testing stateful programs using QuickCheck
+---
+
+When the first version of
+[`QuickCheck`](https://hackage.haskell.org/package/QuickCheck) was released for
+Haskell it was the state-of-the-art in testing. Today however it's lagging
+behind, for example, Erlang's [`PropEr`](https://github.com/manopapad/proper)
+and [`eqc`](http://quviq.com/documentation/eqc/) libraries. The
+[`quickcheck-state-machine`](https://github.com/advancedtelematic/quickcheck-state-machine)
+library is an attempt to add state machine modelling to Haskell's QuickCheck for
+testing stateful/monadic code, and thereby catch up with the Erlang versions of
+QuickCheck.
+
+This proposal is about using, and possibly extending, `quickcheck-state-machine`
+in order to improve the quality of Haskell code in general and for a specific
+project in particular.
+
+The intermediate candidate could:
+
+  1) Find an interesting Haskell library or application to test;
```

Comment:
I would add something along the lines of:
1. It should be a commonly used library
2. The library should be stateful

Reply:
Fixed, thanks!